server/routes/api/team.ts

```diff
@@ -69,4 +72,80 @@ router.post(
   }
 );
 
+router.post("teams.create", auth(), async (ctx) => {
+  const { user } = ctx.state;
+  const { name } = ctx.body;
+
+  const existingTeam = await Team.scope("withAuthenticationProviders").findByPk(
+    user.teamId
+  );
+
+  authorize(user, "createTeam", existingTeam);
+
+  const authenticationProviders = existingTeam?.authenticationProviders.map(
+    (provider) => {
+      return {
+        name: provider.name,
+        providerId: provider.providerId,
+      };
+    }
+  );
+
+  invariant(
+    authenticationProviders?.length,
+    "authentication providers must exist"
+  );
+
+  const team = await sequelize.transaction(async (transaction) => {
+    const team = await teamCreator({
+      name,
+      subdomain: name,
+      authenticationProviders,
+      ip: ctx.ip,
+      transaction,
+    });
+
+    const newUser = await User.create(
+      {
+        teamId: team.id,
+        name: user.name,
+        email: user.email,
+        isAdmin: true,
+        avatarUrl: user.avatarUrl,
+      },
+      { transaction }
+    );
+
+    await Event.create(
+      {
+        name: "users.create",
+        actorId: user.id,
+        userId: newUser.id,
+        teamId: newUser.teamId,
+        data: {
+          name: newUser.name,
+        },
+        ip: ctx.ip,
+      },
+      { transaction }
+    );
+
+    return team;
+  });
+
+  const newUser = await User.findOne({
```

Comment:
Do you remember why you re-query the user that was just created above?

Reply:
pretty sure it was because of the previous way I did transaction passing